### PR TITLE
feat(core-flows, medusa): create sales channel flow

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- create `SalesChannel` workflow and integration with the core

**Depends on**
- [ ] https://github.com/medusajs/medusa/pull/5923

### CODE REVIEW

**Review by olivermrbl — APPROVED** [review]
Niiice!

File: packages/core-flows/src/definition/sales-channel/create-sales-channels.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { createWorkflow } from "@medusajs/workflows-sdk"
+import { CreateSalesChannelDTO, SalesChannelDTO } from "@medusajs/types"
+
+import { createSalesChannelsStep } from "../../handlers/sales-channel"
+
+type WorkflowInput = {
+  channelsData: CreateSalesChannelDTO[]
```
Comment (adrien2p) [inline packages/core-flows/src/definition/sales-channel/create-sales-channels.ts]: suggestion(nit): salesChannelsData

File: packages/core-flows/src/handlers/sales-channel/create-sales-channels.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import {
+  CreateSalesChannelDTO,
+  ISalesChannelModuleService,
+  SalesChannelDTO,
+} from "@medusajs/types"
+
+export const createSalesChannelsStep = createStep<
+  CreateSalesChannelDTO[],
+  SalesChannelDTO[],
+  string[]
```
Comment (adrien2p) [inline packages/core-flows/src/handlers/sales-channel/create-sales-channels.ts]: todo: this should be removed, any particular reason to do that and not rely on the inferece?

File: packages/core-flows/src/handlers/sales-channel/create-sales-channels.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      createdChannels,
+      createdChannels.map((channel) => channel.id)
+    )
+  },
+  async (createdChannelsIds, { container }) => {
+    const salesChannelService: ISalesChannelModuleService = container.resolve(
+      ModuleRegistrationName.SALES_CHANNEL
+    )
+
+    if (createdChannelsIds?.length) {
```
Comment (adrien2p) [inline packages/core-flows/src/handlers/sales-channel/create-sales-channels.ts]: todo: early return here
```suggestion
    if (!createdChannelsIds?.length) {
      return
    }
    
    const salesChannelService: ISalesChannelModuleService = container.resolve(
      ModuleRegistrationName.SALES_CHANNEL
    )
    await salesChannelService!.delete(createdChannelsIds)
```

**Review by adrien2p — APPROVED** [review]
LGTM, few todo's though

**Review by adrien2p — APPROVED** [review]
LGTM

**Pull request closed without merging**